core/src/main/java/com/linecorp/armeria/client/DefaultWebClient.java

```diff
@@ -40,9 +41,29 @@
 
     static final WebClient DEFAULT = new WebClientBuilder().build();
 
-    DefaultWebClient(ClientBuilderParams params, HttpClient delegate,
-                     MeterRegistry meterRegistry, SessionProtocol sessionProtocol, Endpoint endpoint) {
-        super(params, delegate, meterRegistry, sessionProtocol, endpoint);
+    private final Function<? super Endpoint, ? extends EndpointGroup> endpointRemapper;
+
+    DefaultWebClient(ClientBuilderParams params, HttpClient delegate, MeterRegistry meterRegistry) {
+        super(fillDefaultOptions(params), delegate, meterRegistry);
+        endpointRemapper = options().get(WebClientOptions.ENDPOINT_REMAPPER);
+    }
+
+    private static ClientBuilderParams fillDefaultOptions(ClientBuilderParams params) {
+        final ClientOptions options = params.options();
+        if (options.getOrNull(WebClientOptions.ENDPOINT_REMAPPER) != null) {
+            return params;
+        }
+
+        final ClientOptionsBuilder optionsBuilder = options.toBuilder();
+        optionsBuilder.option(WebClientOptions.ENDPOINT_REMAPPER, Function.identity());
```

Comment:
Can we just set this to `ClientOptions.DEFAULT_OPTIONS` so that we don't have to call this method?

Reply:
This option is applicable only to `WebClient`, so I think it doesn't belong there, like gRPC options are not there.

Reply:
Just realized this option mght be applicable for any clients. Let me play with this idea a little bit.

Reply:
Moved `ENDPOINT_MAPPER` to `ClientOptions` and it looks better. This will be useful to non-`WebClient`s when:

- A user is given with a client that talks to a single `Endpoint` and wants to override that.
- We provide a way to send a request to an arbitrary `Endpoint` by using a single client instance in the future.